framework/src/base/Assembly.C

```diff
@@ -1392,6 +1419,15 @@ Assembly::addJacobian(SparseMatrix<Number> & jacobian)
       if ((*_cm)(ivar->number(), jvar->number()) != 0 && _jacobian_block_used[ivar->number()][jvar->number()])
         addJacobianBlock(jacobian, jacobianBlock(ivar->number(), jvar->number()), ivar->dofIndices(), jvar->dofIndices(), ivar->scalingFactor());
 
+ std::vector<dof_id_type> global_indices;
+ for (unsigned int i=0; i<_dof_map.n_dofs(); i++)
+   global_indices.push_back(i);
```

Comment:
I'm sure there are better ways to set this vector. For one resize it fist (upon creation!), then use `[]` to populate it. `push_back` is slow.